### ISSUE TITLE
Demo Level Concept Doc GameGuru MAX - Impressive HUB demo for new users to match expectations

### ISSUE THREAD

**Issue by LeeBamberTGC** [TheGameCreators/GameGuruRepo#5449]
Attached is a proposed design for an improved demo game for the HUB:
[Demo Level Concept Doc GameGuru MAX.pdf](https://github.com/TheGameCreators/GameGuruRepo/files/14947756/Demo.Level.Concept.Doc.GameGuru.MAX.pdf)

**Comment by AvengingEagle** [comment]
I agree with almost all these features, and many have been requested by the community in the past. With one exception: please no roly-polying enemies. Vaulting cover, yes? Rolling about, no. No one does this in combat, it looks silly.

And, as if your list wasn't extensive enough, please consider:
- Making it possible to assign particle effects to weapon models (preferably at a per-bone level). This would be helpful in both FPS contexts where guns could emit smoke, energy weapons could emit sparks, and in an RPG context where we could have magic spells visualised by particle effects, and have flaming torches.
- LUA text formatting commands. I know it doesn't sound sexy, but not having these features currently makes displaying text on screen very basic unless you're prepared to make a sprite for every prompt you want to show.
- Refraction/distortion maps on glass and transparent objects. It makes windows look so much more eye-catching.

Who will make this showcase level? Will it be made internally, or will you be leaning on a member of the community to create it?

**Comment by fearlesswee** [comment]
This all sounds very good, and many of these features proposed are things I've been personally asking and hoping for, for 5+ years at this point, so naturally I'm very excited to see where this goes, and hope all of the new features are added, and sometime within the year.

I would also be more than happy to freely provide any assistance needed to help create this demo level and it's tied new engine features; I have experience with both singleplayer and multiplayer map/level design, .Lua scripting, and designing particles. (I even won that mapping contest MAX had much earlier in it's lifecycle 😉 )

I also agree with AE that the combat rolls from Classic look...silly. Not just from a realism angle, but also the fact that it removes an NPC from being able to look around, shoot, etc. for a moment while they do this awkward roll, which isn't ideal for gameplay where you want your NPCs to engage the player consistently. I don't think they should return for MAX. 

On that note, there's a number of other improvements I think could be made to the AI to make them more engaging and challenging to fight, some of which are already mentioned in the proposal document. 

#4238
 
MAX's AI seem to be in this weird limbo state where they are sluggish to respond to the player, are very hesitant to actually start shooting, spend a lot of time running around aimlessly or just kinda standing there, but when they do _finally_ engage the player, they will melt them with terminator-like accuracy and damage. An ideal enemy AI would consistently engage with the player, but wouldn't deal such extreme damage nor have such extreme accuracy such that the player can realistically deal with them; being constantly engaging so the player has to stay on their toes, but not be so lethal that this consistent engagement is unreasonable or overwhelming.
